### PR TITLE
feat: Narrow down included resources to the allowed resource types

### PR DESCRIPTION
**Pull request by carvantes**
Description of changes:

Narrow down the search results to the allowed resource types as described in https://github.com/awslabs/fhir-works-on-aws-interface/pull/16

**Note**: Need to publish the updated interface to npm first. Until then, checks are failing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/elasticSearchService.test.ts [hunk 1]
```diff
@@ -63,6 +76,7 @@ describe('typeSearch', () => {
                 resourceType: 'Patient',
                 baseUrl: 'https://base-url.com',
                 queryParams,
+                allowedResourceTypes: ALLOWED_RESOURCE_TYPES,
```
Comment (rsmayda) [inline src/elasticSearchService.test.ts:79]: Could we have 1 test with this being an empty array?

File: src/elasticSearchService.ts [hunk 2]
```diff
@@ -258,7 +258,13 @@ export class ElasticSearchService implements Search {
...
             iterative,
         );
-        const { hits } = await this.executeQueries([...includeSearchQueries, ...revIncludeSearchQueries]);
+
+        const lowerCaseAllowedResourceTypes = new Set(request.allowedResourceTypes.map(r => r.toLowerCase()));
+        const allowedInclusionQueries = [...includeSearchQueries, ...revIncludeSearchQueries].filter(query =>
+            lowerCaseAllowedResourceTypes.has(query.index),
+        );
+
+        const { hits } = await this.executeQueries(allowedInclusionQueries);
```
Comment (rsmayda) [inline src/elasticSearchService.ts:267]: FYI just verified we are not breaking from the FHIR standard :)
>... Systems responding to queries might suppress records or may adjust the content of resource instances to exclude data elements the querying system is not permitted to see.

https://www.hl7.org/fhir/updates.html#7.14.3

Comment (carvantes) [reply src/elasticSearchService.ts:267]: There's more specific guidance on https://www.hl7.org/fhir/security.html

> _include and _revinclude search parameters allow client to request related resources. A security implementation must determine if the client has access to the included resources.

File: src/elasticSearchService.test.ts (hunk 1 above)
Comment (carvantes) [reply src/elasticSearchService.test.ts:79]: Added a test for it (Not with an empty array though but rather with a single type on it. At the very least you are always allowed to search for the `TypeSearchRequest.resourceType `)

**Review by rsmayda — APPROVED** [review]

**Pull request merged**